File: src/protocol/node/packet/DataType.ts
/*
 * Filename: DataType.ts
 * FullPath: apps/CWSP-reborn/src/protocol/node/packet/DataType.ts
 * Change date and time: 17.20.00_10.07.2026
 * Reason for changes: Thin NodeJS facade re-exporting the unified DataAsset envelope helper. | Alias migration 17.20.00_10.07.2026: long relative cwsp-shared imports -> @fest-lib/cwsp-shared/v2/*.
 *
 * NOTE: Re-export only. Binary/file/blob/base64/url normalization stays single-source in
 * `@fest-lib/cwsp-shared` v2 (see features-data-asset.mdc).
 */

export { normalizeDataAssetEnvelope } from "@fest-lib/cwsp-shared/v2/validation.ts";

export type { DataAssetEnvelope } from "@fest-lib/cwsp-shared/v2/types.ts";

File: src/protocol/java/packet/DataType.java
/*
 * Filename: DataType.java
 * FullPath: apps/CWSP-reborn/src/protocol/java/packet/DataType.java
 * Change date and time: 16.42.00_10.07.2026
 * Reason for changes: DataAsset source discriminator constants (mirrors the
 *                     {@code source} field values of DataAssetEnvelope).
 */
package space.u2re.cwsp.protocol.packet;

/** DataAsset {@code source} values used by the unified DataAsset pipeline. */
public final class DataType {

    public static final String DATA_URL = "data-url";
    public static final String BASE64 = "base64";
    public static final String URL = "url";
    public static final String FILE = "file";
    public static final String BLOB = "blob";
    public static final String TEXT = "text";

    private DataType() {
    }
}
